subsys/logging/log_core.c

```diff
@@ -182,24 +182,11 @@ static void detect_missed_strdup(struct log_msg *msg)
 #undef ERR_MSG
 }
 
-static inline void msg_finalize(struct log_msg *msg,
-				struct log_msg_ids src_level)
+void z_log_msg_post_finalize(void)
 {
-	unsigned int key;
-
-	msg->hdr.ids = src_level;
-	msg->hdr.timestamp = timestamp_func();
-
 	atomic_inc(&buffered_cnt);
-
-	key = irq_lock();
-
-	log_list_add_tail(&list, msg);
-
-	irq_unlock(key);
-
 	if (panic_mode) {
-		key = irq_lock();
+		int key = irq_lock();
```

Comment:
Was this change from `unsigned int` to `int` intentional, or should we preserve it?

Reply:
@jenmwms check API definition for `irq_lock()` to see its return value type

Reply:
@andrewboie Thanks! So, it needs to be `unsigned int`, from [Interrupt Service Routine APIs: irq_lock( )](https://docs.zephyrproject.org/apidoc/latest/group__isr__apis_ga19fdde73c3b02fcca6cf1d1e67631228.html#ga19fdde73c3b02fcca6cf1d1e67631228)

` It returns an unsigned integer "lock-out key",`

Reply:
fixed